src/components/Modal/Modal.vue

```diff
@@ -40,15 +40,21 @@
     </template>
 
     <template #header>
-      <div v-if="icon" :class="`modal-icon-container modal-${iconColor}`">
-        <Icon :icon="icon" :size-override="35" />
+      <div
+        v-if="icon || slots.icon"
+        :class="`modal-icon-container modal-${iconColor}`"
+      >
+        <Icon v-if="icon" :icon="icon" :size-override="35" />
+        <slot name="icon" :size-override="35" />
```

Comment:
how is this handled if an icon is passed both as a prop and within a slot?

Reply:
They will both be rendered, I went back and forth on which should take priority but I thought that might get confusing vs. just showing if you put them on there